Sources/SpeziAccount/AccountNotifications.swift

```diff
@@ -26,19 +26,36 @@ import Spezi
 public final class AccountNotifications {
     /// Describes an Account event.
     public enum Event {
+        /// A new account was associated due to a login or signup operation.
+        case didAssociate(_ details: AccountDetails)
+        /// The details of the currently associated Account changed.
+        case detailsChanged(_ previous: AccountDetails, _ new: AccountDetails)
```

Comment:
Maybe we should also name this: didChangeDetails to be aligned with the other naming?
```suggestion
        case didChangeDetails(_ previous: AccountDetails, _ new: AccountDetails)
```